File: apps/api-crm/Models/Campaign.cs
namespace Crm.Api.Models;

public class Campaign
{
    public Guid Id { get; set; }
    public string Title { get; set; } = string.Empty;
    public string Subject { get; set; } = string.Empty;
    public string Description { get; set; } = string.Empty;
    public List<string> Channels { get; set; } = []; // Email, InApp, Facebook, Twitter, Instagram
    public string TargetAudience { get; set; } = "All"; // All, Regular, InstitutionalBuyer, Specific
    public List<string>? TargetCustomerIds { get; set; } // Specific CustomerProfile IDs if TargetAudience == "Specific"
    public List<string>? TargetEmails { get; set; } // Explicit email addresses if typed by staff
    public string Status { get; set; } = "Draft"; // Draft, Active, Ended
    public Guid? TemplateId { get; set; }
    public string? ImageUrl { get; set; }
    public string CreatedById { get; set; } = string.Empty;
    public DateTime CreatedAt { get; set; }

    // Navigation properties
    public User CreatedBy { get; set; } = null!;
    public Template? Template { get; set; }
    public CampaignSchedule? CampaignSchedule { get; set; }
    public ICollection<MarketingInteraction> MarketingInteractions { get; set; } = [];
}

File: apps/web-crm/src/types/campaign.ts
export type CampaignStatus = "Draft" | "Active" | "Ended";
export type CampaignChannel = "Email" | "InApp";
export type ScheduleType = "SendNow" | "Scheduled" | "Recurring";
export type RecurrenceDay = "Monday" | "Tuesday" | "Wednesday" | "Thursday" | "Friday" | "Saturday" | "Sunday";

export interface CampaignSchedule {
  scheduleType: ScheduleType;
  recurrenceDays?: RecurrenceDay[];
  startDate?: string | null;
  endDate?: string | null;
}

export interface CampaignListItem {
  id: string;
  title: string;
  channels: CampaignChannel[];
  targetAudience?: string;
  targetCustomerIds?: string[];
  targetEmails?: string[];
  status: CampaignStatus;
  createdAt: string;
}

export interface Campaign extends CampaignListItem {
  subject: string;
  description: string;
  templateId?: string | null;
  imageUrl?: string | null;
  createdById?: string;
  schedule?: CampaignSchedule | null;
}

export interface CreateCampaignInput {
  title: string;
  subject: string;
  description: string;
  channels: CampaignChannel[];
  targetAudience?: string;
  targetCustomerIds?: string[];
  targetEmails?: string[];
  scheduleType: ScheduleType;
  recurrenceDays?: RecurrenceDay[];
  startDate?: string;
  endDate?: string;
  imageUrl?: string;
  templateId?: string;
  status?: CampaignStatus;
}

export interface UpdateCampaignInput {
  title?: string;
  subject?: string;
  description?: string;
  channels?: CampaignChannel[];
  targetAudience?: string;
  targetCustomerIds?: string[];
  targetEmails?: string[];
  scheduleType?: ScheduleType;
  recurrenceDays?: RecurrenceDay[];
  startDate?: string;
  endDate?: string;
  imageUrl?: string;
  templateId?: string;
  status?: CampaignStatus;
}

export interface Template {
  id: string;
  name: string;
  description?: string | null;
  contentHtml: string;
  thumbnailUrl?: string | null;
  channel: string;
  createdAt: string;
}

export interface CampaignDispatchResult {
  totalRecipients: number;
  sentCount: number;
  failedCount: number;
  errors: string[];
  message: string;
}

